examples/insights/aws-cost/handler.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+const dateFormat = require('dateformat');
+const rp = require('request-promise');
+var AWS = require("aws-sdk");
+
+function validateInputBody(inputBody) {
+    if (!inputBody.username) {
+        return ({status: false, message: 'Please enter username for F5 Cloud Services Account. Eg, "username": "foo"'});
+    } else if (!inputBody.password) {
+        return ({
+            status: false,
+            message: 'Please enter password for F5 Cloud Services Account. Eg, "password": "fooPassword"'
+        });
+    } else if (!inputBody.accountid) {
+        return ({
+            status: false,
+            message: 'Please enter Account-ID for Preferred F5 Cloud Services Account. Eg, "accountid": "a-fooAccountID"'
+        });
+    }
+    return ({status: true, message: "All input parameters received."})
+}
+
+module.exports.costInsight = async event => {
+    const self = this;
+
+    function getAccessKey(Username, Password) {
+        return new Promise((resolve) => {
+            resolve(
+                rp.post({
+                    url: 'https://api.dev.f5aas.com/v1/svc-auth/login',
+                    body: {username: Username, password: Password},
+                    json: true
+                }));
+        });
+    }
+
+    function loginF5Portal(AccessToken) {
+        return new Promise((resolve) => {
+            resolve(
+                rp.defaults({
+                    auth: {bearer: AccessToken},
+                    baseUrl: "https://api.dev.f5aas.com/beacon/v1/",
+                    json: true
+                })
+            );
+        });
+    }
+
+    function getCost(StartDate, EndDate) {
+        var costexplorer = new AWS.CostExplorer();
+
+        // Documentation: https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/ce-exploring-data.html
+        var costParams = {
+            TimePeriod: {
+                End: EndDate,
+                Start: StartDate
+            },
+            Granularity: 'MONTHLY',
+            Metrics: ['AmortizedCost']
+        };
+
+        return new Promise((resolve, reject) => {
+            costexplorer.getCostAndUsage(costParams, function (err, data) {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve(data);
+                }
+            })
+        });
+    }
+
+    function createInsight(CostArray) {
```

Comment:
nit: seems this function is not doing and API request. As it is just a data transformation, promise is not needed for this.